services/core-web/src/components/Forms/permits/PreDraftPermitForm.js

```diff
@@ -1,23 +1,73 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
+import { compose, bindActionCreators } from "redux";
+import { connect } from "react-redux";
 import PropTypes from "prop-types";
-import { Field, reduxForm } from "redux-form";
+import { Field, reduxForm, getFormValues, change } from "redux-form";
 import { Form } from "@ant-design/compatible";
 import "@ant-design/compatible/assets/index.css";
-import { Col, Row } from "antd";
+import { Col, Row, Tooltip } from "antd";
 import { required, requiredRadioButton } from "@common/utils/Validate";
 import { resetForm, createDropDownList } from "@common/utils/helpers";
 import * as FORM from "@/constants/forms";
 import { renderConfig } from "@/components/common/config";
 import CustomPropTypes from "@/customPropTypes";
+import { getDropdownPermitAmendmentTypeOptions } from "@common/selectors/staticContentSelectors";
+import { PERMIT_AMENDMENT_TYPES } from "@common/constants/strings";
 
 const propTypes = {
   isAmendment: PropTypes.bool.isRequired,
   permits: PropTypes.arrayOf(CustomPropTypes.permit).isRequired,
   isCoalOrMineral: PropTypes.bool.isRequired,
+  permitAmendmentTypeDropDownOptions: CustomPropTypes.options.isRequired,
+  change: PropTypes.func.isRequired,
 };
 
 export const PreDraftPermitForm = (props) => {
+  const [permitType, setPermitType] = useState(PERMIT_AMENDMENT_TYPES.original);
+
+  useEffect(() => {
+    if (!props.isAmendment) {
+      props.change("permit_amendment_type_code", permitType);
+    }
+  });
+
+  const getPermitType = (selectedPermitGuid) => {
+    if (props.permits && props.permits.length > 0) {
+      const selectedPermit = props.permits.find(
+        (permit) => permit.permit_guid === selectedPermitGuid
+      );
+      if (selectedPermit.permit_amendments && selectedPermit.permit_amendments.length > 0) {
+        const selectedPermitType =
+          selectedPermit.permit_amendments.filter(
+            (a) => a.permit_amendment_type_code === PERMIT_AMENDMENT_TYPES.amalgamated
+          ).length > 0
+            ? PERMIT_AMENDMENT_TYPES.amalgamated
+            : PERMIT_AMENDMENT_TYPES.amendment;
+        setPermitType(selectedPermitType);
+        props.change("permit_amendment_type_code", selectedPermitType);
+      }
+    }
+  };
+
   const permitDropdown = createDropDownList(props.permits, "permit_no", "permit_guid");
+  let tooltip = "";
+  let isPermitAmendmentTypeDropDownDisabled = true;
+  let permitAmendmentDropdown = props.permitAmendmentTypeDropDownOptions;
+
+  if (permitType === PERMIT_AMENDMENT_TYPES.amalgamated) {
+    tooltip = "You can issue only amalgamated permits";
+  }
+  if (permitType === PERMIT_AMENDMENT_TYPES.original) {
```

Comment:
Can you move all this logic into a function?

Reply:
discussed in call and decided to leave as it is because it is relevant for the render only